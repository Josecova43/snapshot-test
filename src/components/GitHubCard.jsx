import { Card } from "react-bootstrap"
import { useState, useEffect } from "react"

export default function GitHubCard() {
    const [githubData, setGithubData]= useState({
        name:'',
        avatar_url: '',
        bio: '',
        });

        useEffect( async () => {
             await fetch(`https://api.github.com/users/Josecova41`)
            .then(response => response.json())
            .then(data => setGithubData(data))

        });



    return (
        <div>
            <Card>
                <Card.Body>
                    <Card.Img 
                    variant="top"
                    src={githubData.avatar_url}
                    alt="Github avatar"
                    />

                    <Card.Title>
                        <h2>{githubData.name}       </h2>
                    </Card.Title>
                    <Card.Text>
                        {
                            githubData.bio
                        }
                    </Card.Text>
                </Card.Body>
            </Card>

        </div>

    )
}