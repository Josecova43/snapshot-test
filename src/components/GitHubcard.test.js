import {render, screen, waitFor } from '@testing-library/react'
import renderer from 'react-test-renderer'
import GitHubCard from './GitHubCard'


 const DEFAULT_MOCK_DATA ={
    name: 'jose',
    avatar_url: 'https://avatars.githubusercontent.com/u/116780029?v=4',
    bio: 'Pretty cool guy '
}

const DEFAULT_AVATAR_ALT= 'Github avatar'

describe('Test GitHUbCard for features when', () =>{
    beforeEach(()=>{
     
        fetch.mockResponseOnce(JSON.stringify(DEFAULT_MOCK_DATA))
        render(<GitHubCard />)
    })
    afterEach(()=> {
        fetch.resetMocks();
    })

    test('it contains a photo of the github user', async () => {
        const avatarImage= await waitFor( () => screen.getByAltText('Github avatr'));
        expect(avatarImage).toHaveAttribute('src', 
        expect.stringContaining(DEFAULT_MOCK_DATA.avatar_url))
    });

    test('it contain a name of the gthub user', async () => {
        const userName = await waitFor(()=>screen.getByRole('heading', {level: 2}));
        expect(userName).toHaveTextContent(DEFAULT_MOCK_DATA.name)

    });


    test('it contain a blurb of the gthub user', async () => {
        const blurb = await waitFor(() => screen.getByRole('paragraph'));
        expect(blurb).toHaveTextContent(DEFAULT_MOCK_DATA.bio)

    });

    test('it renders a snapshot',  () => {
        const tree = renderer.create(<GitHubCard />).toJSON();
        expect(tree).toMatchSnapshot();

    });
})