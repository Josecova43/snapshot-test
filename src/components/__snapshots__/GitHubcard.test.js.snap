// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GitHUbCard for features when it renders a snapshot 1`] = `
<div>
  <div
    className="card"
  >
    <div
      className="card-body"
    >
      <img
        alt="Github avatar"
        className="card-img-top"
        src=""
      />
      <div
        className="card-title h5"
      >
        <h2>
                 
        </h2>
      </div>
      <p
        className="card-text"
      />
    </div>
  </div>
</div>
`;
